zebra-chain/src/orchard/note/nullifiers.rs

```diff
@@ -35,14 +37,27 @@ fn prf_nf(nk: pallas::Base, rho: pallas::Base) -> pallas::Base {
 }
 
 /// A Nullifier for Orchard transactions
-#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Copy, Debug, Eq, Serialize, Deserialize)]
 pub struct Nullifier(#[serde(with = "serde_helpers::Base")] pallas::Base);
 
+impl Hash for Nullifier {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.0.to_bytes().hash(state);
+    }
+}
+
 impl From<[u8; 32]> for Nullifier {
     fn from(bytes: [u8; 32]) -> Self {
         Self(pallas::Base::from_bytes(&bytes).unwrap())
     }
 }
+// Clippy is failing if we derive `PartialEq` with implemented `Hash`
+// error: you are implementing `Hash` explicitly but have derived `PartialEq`
+impl PartialEq for Nullifier {
```

Comment:
We don't really need this comment in the code.

Any PR that tries to derive the `PartialEq` impl will fail the clippy CI check.
```suggestion

impl PartialEq for Nullifier {
```